src/stmem.h

```diff
@@ -20,19 +20,26 @@ class mem_hdr
 {
 
 public:
-    mem_hdr();
+    /// \param locked whether the associated StoreEntry is locked
+    mem_hdr(bool locked);
     ~mem_hdr();
     void freeContent();
     int64_t lowestOffset () const;
     int64_t endOffset () const;
     int64_t freeDataUpto (int64_t);
     ssize_t copy (StoreIOBuffer const &) const;
     bool hasContigousContentRange(Range<int64_t> const &range) const;
-    /* success or fail */
+    /// Saves the buffer into the internal storage.
+    /// Do not call directly - use StoreEntry::writeData() instead.
     bool write (StoreIOBuffer const &);
```

Comment:
The disadvantage of this approach is that calling mem_hdr::write() would be disastrous - the counter will not be properly updated leading eventually to an assertion. Alternatively, we could pass StoreEntry* pointer as an argument and calculate needsIdlePagesUpdating() directly here.